rfd/0241-scoped-access-lists.md

```diff
@@ -0,0 +1,931 @@
+roles at specific scopes.
+
+Required Reading:
+
+- [RFD 229 - Scopes](./0229-scopes.md)
+
+Recommended Reading:
+
+- [RFD 6e - Access Lists](https://github.com/gravitational/teleport.e/blob/master/rfd/0006e-access-lists.md)
+- [RFD 170 - Nested Access Lists](./0170-nested-accesslists.md)
+
+Scoped access lists will initially aim to provide a small subset of the
+functionality of traditional access lists, tailored for the specific use case
+of using scoped access lists to manage scoped privileges in the context of
+automation and identity provider sync.
+
+## Why
+
+Where reasoning about users in terms of groups or lists is common and natural,
+the main goal of scoped access lists is to allow groups of users to be defined
+such that all members of those user groups can be granted a common set of
+scoped privileges.
+
+These user groups (access lists and their members) will be able to be defined
+via automation such as terraform, or synced from an external identity provider.
+
+In the current state of the scopes implementation, there is no way to assign
+scoped roles to groups of users, admins must create a scoped role assignment
+for every (user, scoped_role) pair.
+
+With scoped access lists, admins will define a scoped access list for each
+logical group of users.
+The scoped access list definition will include the set of scoped roles that
+should be granted to all members of that access list.
+Then access list memberships may be synced from an identity provider, defined in
+Terraform, or created with `tctl`.
+
+## Details
+
+### UX
+
+#### User story
+
+Examplecorp operates out of two regions, east and west.
+Each region has their own groups of admins and users.
+They want to grant each admin group broad administrative privileges over
+everything in their region, and each user group access to everything in their
+region, without allowing admins or users in each region to modify or access
+anything in the other region (due to the ongoing east vs west rivalry).
+
+They have defined the following scoped roles that they want to grant to their users:
+
+```yaml
+kind: scoped_role
+metadata:
+  name: ops-admin
+scope: /ops
+spec:
+  assignable_scopes:
+  - /ops/**
+  rules:
+  - resources:
+    - scoped_role
+    - scoped_role_assignment
+    - scoped_access_list
+    - scoped_access_list_member
+    verbs: [create, list, readnosecrets, update, delete]
+---
+kind: scoped_role
+metadata:
+  name: ops-staging-access
+scope: /ops
+spec:
+  assignable_scopes:
+  - /ops/**
+  node_labels:
+  - name: 'env'
+    values: ['staging']
+  logins: ['opsuser', 'root']
+---
+kind: scoped_role
+metadata:
+  name: ops-prod-access
+scope: /ops
+spec:
+  assignable_scopes:
+  - /ops/**
+  node_labels:
+  - name: 'env'
+    values: ['prod']
+  logins: ['opsuser']
+```
+
+But now they are faced with the challenge of assigning these roles to all their
+thousands of users.
+They have all their users and groups already defined in Terraform, but they
```

Comment:
How currently Teleport allow to  define a Group in terraform ? 

Almost all of marked is using  Either Okta or Enta to manage users and groups.